test/blackbox-tests/test-cases/dune-preprocess.t/run.t

```diff
@@ -0,0 +1,4 @@
+This test check that we can show the preprocessed output of a source code
+
+  $ dune preprocess src/main.ml
```

Comment:
Sorry to butt in, but while I don't disagree at all with wanting to limit the number of toplevel subcommands, from a discoverability perspective I would not think to look for preprocessing under `describe`. Perhaps I have not understood as intended, but my interpretation of `describe` is to provide information that is specified in the build configuration, or derived from it, but not really information that comes from processing the source code itself.

Reply:
I don't have a strong opinion so I tag the people in the original issue. @smorimoto @bobot @jeremiedimino thoughts?

Reply:
I suppose a better name would be `dune print ast` or something.

> I would not think to look for preprocessing under describe. Perhaps I have not understood as intended, but my interpretation of describe is to provide information that is specified in the build configuration, or derived from it, but not really information that comes from processing the source code itself.

While that's true, I also don't like introducing a redundant toplevel name like `print` for this. At the end of the day, `describe` prints information in either machine or human readable form. The source of this information does not seem like an important thing to distinguish things by.

Reply:
Well, it indeed does seem redundant to add subcommands like `print`, `expand` or `preprocess`, and I don't have a strong opinion on this, so I think you can move things forward in the "dune way", but if you're going to do that you need more specific guidance on the best command name in the "dune way".

Reply:
I think `pp` or `ppxed` would be good. Generally, `pp` is one of the most common pieces of the OCaml codebase, but that doesn't mean it's very accessible.

Reply:
@jberdine @rgrinberg what's your opinion about
* `dune show pp`,
* `dune show ppxed`,
* `dune print pp`, or
* `dune print ppxed`?

I would like to go with the one you prefer, or `dune show ppxed` if all seems equally good for you.

Reply:
We decided to go with `dune describe pp`. The current description of `describe` fits this use very well:

```
Describe what is in the current workspace in either human or
        machine readable form.

        By default, this command output a human readable description of
        the current workspace. This output is aimed at human and is not
        suitable for machine processing. In particular, it is not versioned.

        If you want to interpret the output of this command from a program,
        you must use the $(b,--format) option to specify a machine readable
        format as well as the $(b,--lang) option to get a stable output.
```

In particular, `--lang` restrictions are not applicable since the output is indeed human readable.